### PR TITLE
Feature/compatebility context

### PR DESCRIPTION
**Pull request by AlonZivony**
## Initial Checklist

- [ ] There is an issue describing the need for this PR.
- [ ] Git log contains summary of the change.
- [ ] Git log contains motivation and context of the change.
- [ ] If part of an EPIC, PR git log contains EPIC number.
- [ ] If part of an EPIC, PR was added to EPIC description.

### types: add compatibility mode context flag
Add a flag that signifies if the process is running in 32bit
compatibility mode for each event.
This will help to understand better process context and improve other
features in the future.

### events: pass compatibility flag in eBPF events
Pass a flag if the process run in 32bit compatibility mode in eBPF
context.

Fixes: #2613 

## Type of change

- [ ] Bug fix (non-breaking change fixing an issue, preferable).
- [ ] Quick fix (minor non-breaking change requiring no issue, use with care)
- [ ] Code refactor (code improvement and/or code removal)
- [ ] New feature (non-breaking change adding functionality).
- [ ] Breaking change (cause existing functionality not to work as expected).

## How Has This Been Tested?

Tests being included in this PR:

- [ ] Test File A
- [ ] Test File B

Reproduce the test by running:

- command 01
- command 02

## Final Checklist:

Pick "Bug Fix" or "Feature", delete the other and mark appropriate checks.

- [ ] I have made corresponding changes to the documentation.
- [ ] My code follows the style guidelines (C and Go) of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented all functions/methods created explaining what they do.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix, or feature, is effective.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published before.

## Git Log Checklist:

My commits logs have:

- [ ] Subject starts with "subsystem|file: description".
- [ ] Do not end the subject line with a period.
- [ ] Limit the subject line to 50 characters.
- [ ] Separate subject from body with a blank line.
- [ ] Use the imperative mood in the subject line.
- [ ] Wrap the body at 72 characters.
- [ ] Use the body to explain what and why instead of how.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -101,3 +101,5 @@ require (
 )
 
 replace github.com/kubernetes/cri-api => k8s.io/cri-api v0.23.5-rc.0
+
+replace github.com/aquasecurity/tracee/types => ./types
```
Comment (yanivagman) [inline go.mod:105]: need to remove this replace

File: pkg/ebpf/c/tracee.bpf.c [hunk 2]
```diff
@@ -524,9 +524,11 @@ enum event_id_e
 #define CONT_ID_LEN          12
 #define CONT_ID_MIN_FULL_LEN 64
 
+// Flags in the each task's context
 enum context_flags_e
 {
-    CONTAINER_STARTED_FLAG = (1 << 0)
+    CONTAINER_STARTED_FLAG = (1 << 0), // mark the task's container have started
+    IS_COMPAT_PROC_FLAG = (1 << 1)     // is the task running in compatible mode
```
Comment (yanivagman) [inline pkg/ebpf/c/tracee.bpf.c:531]: IS_COMPAT_FLAG (without PROC) would be a better name since this is per task

File: pkg/ebpf/c/tracee.bpf.c [hunk 3]
```diff
@@ -809,8 +811,9 @@ typedef struct io_data {
 } io_data_t;
 
 typedef struct proc_info {
-    bool new_proc; // set if this process was started after tracee. Used with new_pid filter
-    bool follow;   // set if this process was traced before. Used with the follow filter
+    bool new_proc;  // set if this process was started after tracee. Used with new_pid filter
+    bool follow;    // set if this process was traced before. Used with the follow filter
+    bool is_compat; // is the process running in compatible mode
```
Comment (yanivagman) [inline pkg/ebpf/c/tracee.bpf.c:816]: This variable is not used anywhere

**Review by yanivagman — CHANGES_REQUESTED** [review]

File: go.mod (hunk 1 above)
Comment (AlonZivony) [reply go.mod:105]: Just wanted to show that it passes the tests and work fine.
I will open now PR for the types and update to it when merged.

File: pkg/ebpf/c/tracee.bpf.c [hunk 4]
```diff
@@ -524,9 +524,11 @@ enum event_id_e
 #define CONT_ID_LEN          12
 #define CONT_ID_MIN_FULL_LEN 64
 
+// Flags in the each task's context
 enum context_flags_e
 {
-    CONTAINER_STARTED_FLAG = (1 << 0)
+    CONTAINER_STARTED_FLAG = (1 << 0), // mark the task's container have started
+    IS_COMPATFLAG = (1 << 1)           // is the task running in compatible mode
```
Comment (NDStrahilevitz) [inline pkg/ebpf/c/tracee.bpf.c:531]:
```suggestion
    IS_COMPAT_FLAG = (1 << 1)           // is the task running in compatible mode
```

**Review by NDStrahilevitz — APPROVED** [review]
LGTM, @yanivagman WDYT now?

File: pkg/ebpf/c/tracee.bpf.c [hunk 5]
```diff
@@ -524,9 +524,11 @@ enum event_id_e
 #define CONT_ID_LEN          12
 #define CONT_ID_MIN_FULL_LEN 64
 
+// Flags in the each task's context
```
Comment (yanivagman) [inline pkg/ebpf/c/tracee.bpf.c:527]: typo `in the`

File: pkg/ebpf/events_pipeline.go [hunk 6]
```diff
@@ -318,9 +318,11 @@ func (t *Tracee) shouldProcessEvent(event *trace.Event) bool {
 func parseContextFlags(flags uint32) trace.ContextFlags {
 	const (
 		ContainerStartFlag = 1 << iota
+		IsCompatProc
```
Comment (yanivagman) [inline pkg/ebpf/events_pipeline.go:321]: IsCompatFlag to be consistent with ContainerStartFlag

**Review by yanivagman — APPROVED** [review]
LGTM

**Pull request merged**